packaging/installer/netdata-updater.sh

```diff
@@ -372,16 +372,27 @@ get_netdata_latest_tag() {
 newer_commit_date() {
   info "Checking if a newer version of the updater script is available."
 
+  commit_check_url="https://api.github.com/repos/netdata/netdata/commits?path=packaging%2Finstaller%2Fnetdata-updater.sh&page=1&per_page=1"
+  python_version_check="from __future__ import print_function;import sys,json;data = json.load(sys.stdin);print(data[0]['commit']['committer']['date'] if isinstance(data, list) else '')"
+
   if command -v jq > /dev/null 2>&1; then
-    commit_date="$(_safe_download "https://api.github.com/repos/netdata/netdata/commits?path=packaging%2Finstaller%2Fnetdata-updater.sh&page=1&per_page=1" /dev/stdout | jq '.[0].commit.committer.date' | tr -d '"')"
+    commit_date="$(_safe_download "${commit_check_url}" /dev/stdout | jq '.[0].commit.committer.date' 2>/dev/null | tr -d '"')"
   elif command -v python > /dev/null 2>&1;then
-    commit_date="$(_safe_download "https://api.github.com/repos/netdata/netdata/commits?path=packaging%2Finstaller%2Fnetdata-updater.sh&page=1&per_page=1" /dev/stdout | python -c 'from __future__ import print_function;import sys,json;print(json.load(sys.stdin)[0]["commit"]["committer"]["date"])')"
+    commit_date="$(_safe_download "${commit_check_url}" /dev/stdout | python -c "${python_version_check}")"
   elif command -v python3 > /dev/null 2>&1;then
-    commit_date="$(_safe_download "https://api.github.com/repos/netdata/netdata/commits?path=packaging%2Finstaller%2Fnetdata-updater.sh&page=1&per_page=1" /dev/stdout | python3 -c 'from __future__ import print_function;import sys,json;print(json.load(sys.stdin)[0]["commit"]["committer"]["date"])')"
+    commit_date="$(_safe_download "${commit_check_url}" /dev/stdout | python3 -c "${python_version_check}")"
   fi
 
   if [ -z "${commit_date}" ] ; then
-    commit_date="9999-12-31T23:59:59Z"
+    commit_date="9999999999"
+  elif [ "$(uname)" = "Linux" ]; then
+    commit_date="$(date -d "${commit_date}" +%s)"
+  else # assume BSD-style `date` if we are not on Linux
+    commit_date="$(date -j -f "%Y-%m-%dT%H:%M:%SZ" "${commit_date}" +%s 2>/dev/null)"
+
+    if [ -z "${commit_date}" ]; then
+      commit_date="9999999999"
```

Comment:
The default was to download the newest version when fail to find the last commit date.

I see that `9999999999` (new) < `9999-12-31T23:59:59Z` (old).

Also, I think it would be less error-prone to `[ -z "${commit_date}" ]` just once right before the return

```bash
[ -z "${commit_date}" ] && commit_date=$(date -d "9999-12-31T23:59:59Z" +%s)
[ "${commit_date}" -ge "${script_date}" ]
```

Reply:
An alternative way is an early return, we can `return 1` right after _safe_download + extracting commit date if `[ -z "${commit_date}" ]`.

Reply:
BSD systems do not have the `-d` option, and GNU `date` was actually choking on the old timestamp on at lest some systems (see the output in the original issue), hence the changes here.

An early return though is probably the best solution.

Reply:
Also, we actually do have to check for an empty date twice here.

The first case is for when the API call did not return a usable date. We need this to prevent running the GNU `date` command with an empty argument passed to the `-d` option, because that is the same as not having a `-d` option at all (and thus spits out the _current_ time, which is not what we want).

The second case inside the `else` conditional is for the case where the command does not parse the date correctly (we end up getting nothing on stdout in this case).